conn_linux.go

```diff
@@ -450,14 +439,17 @@ func (s *sysSocket) Socket(family int) error {
 		return err
 	}
 
-	s.fd = fd
+	if err := setBlockingMode(fd); err != nil {
+		return err
+	}
+	s.fd = os.NewFile(uintptr(fd), "netlink")
```

Comment:
I would like to see a comment here noting that passing a non-blocking FD to this function registers it with the poller, and a link to https://golang.org/pkg/os/#NewFile, for future readers.